Framework/Algorithms/src/PDCalibration.cpp

```diff
@@ -1039,6 +1078,7 @@ void PDCalibration::fitDIFCtZeroDIFA_LM(const std::vector<double> &d, const std:
 vector<double> PDCalibration::dSpacingWindows(const std::vector<double> &centres,
                                               const std::vector<double> &windows_in) {
 
+  assert(windows_in.size() == 1 || windows_in.size() / 2 == centres.size());
```

Comment:
`assert` only applies in DEBUG builds. If this is something that should always be checked I recommend switching to `if` with `raise`